protobuf/datastore.proto

```diff
@@ -139,14 +151,33 @@ message PutObsResponse {
 
 //---------------------------------------------------------------------------
 
+message Strings {
+  repeated string values = 1;
+}
+
 message GetObsRequest {
-  TimeInterval interval = 1; // only return observations in this time range
-  Polygon inside = 2; // if specified, only return observations in this area
-  repeated string platforms = 3; // if specified, only return observations matching any of these platform patterns
-  repeated string standard_names = 4 [json_name = "standard_names"]; // if specified, only return observations matching any of these standard names
-  repeated string instruments = 5; // if specified, only return observations matching any of these instruments
-  repeated string processing_levels = 6 [json_name = "processing_levels"]; // if specified, only return observations matching any of these processing levels
-  // TODO: add search filters for other metadata
+  // --- BEGIN non-string metadata -------------------------
+
+  // temporal filter
+  TimeInterval interval = 1; // only observations in this time range may be returned
+
+  // spatial filter
+  Polygon inside = 2; // if specified, only observations in this area may be returned
+
+  // search wrt. TSMetadata.links
+  // TODO - needs special handling
+
+  // --- END non-string metadata -------------------------
+
+  // --- BEGIN string metadata (handleable with reflection) -------------------------
+
+  // general filter
+  // - map keys must correspond exactly with string field names in TSMetadata or ObsMetadata
+  // - if map key F is specified (where F is for example 'platform'), only observations that match
+  //   at least one of these values for F may be returned
+  map<string, Strings> filter = 3;
```

Comment:
Hmm, looking at the Python, what you have now is probably best.

Reply:
I wish it was possible in Python to write just

```
dstore.GetObsRequest(filter=dict(instrument=["tn", ...]))
```
or
```
dstore.GetObsRequest(filter={'instrument':["tn", ...]})
```
instead of
```
dstore.GetObsRequest(filter=dict(instrument=dstore.Strings(values=["tn"])))
```
But I guess one gets used to the more explicit form,